jupyterhub/app.py

```diff
@@ -365,12 +366,33 @@ def _log_format_default(self):
         """override default log format to include time"""
         return "%(color)s[%(levelname)1.1s %(asctime)s.%(msecs).03d %(name)s %(module)s:%(lineno)d]%(end_color)s %(message)s"
 
+    extra_log_file = Unicode("", config=True)
+    extra_log_handler = Instance(
+        klass=logging.Handler,
+        allow_none=True,
+        config=True,
```

Comment:
I'm trying to think if having the instance as a config value is more or less useful than calling:

```python
logging.getLogger('JupyterHub').add_handler(my_handler)
```

Especially since registering directly with the logger doesn't make it mutually exclusive with log-to-file. What do you think?